### PR TITLE
TSA ARMA

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Continues towards #1404 

##### What are the significant changes in functionality due to this change request?
Adds the basic ARMA to TSA algorithm (no VARMA, but including gaussianizing through CDF)

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [ ] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
@alfoa tagging you in case you want to look at this as well

File: framework/TSA/ARMA.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+                   MiB, but increased training time by 0.4 seconds. No change in results has been
+                   observed switching between modes. Note that the ARMA must be
+                   retrained to change this property; it cannot be applied to serialized ARMAs.
+                   \default{False}""")
+    specs.addSub(InputData.parameterInputFactory('SignalLag', contentType=InputTypes.FloatType,
+                 descr=r"""the number of terms in the AutoRegressive term to retain in the
+                           regression; "P" in literature."""))
+    specs.addSub(InputData.parameterInputFactory('NoiseLag', contentType=InputTypes.FloatType,
+                 descr=r"""the number of terms in the Moving Average term to retain in the
+                           regression; "Q" in literature."""))
```
Comment (dylanjm) [inline framework/TSA/ARMA.py:65]: What do you think of adding _"Represented as "Q" in literature."_ or _"Typically Represented as "Q" in literature."_ Just to be a little more explicit?

File: tests/framework/ROM/TimeSeries/SyntheticHistory/TrainingData/generateARMA.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
```
Comment (dylanjm) [inline tests/framework/ROM/TimeSeries/SyntheticHistory/TrainingData/generateARMA.py:1]: Would this file and the two other .csv files go better in the tests folder?

**Review by dylanjm — APPROVED** [review]
I think everything looks pretty good. 

* All the code logic seems to make sense. 
* Required documentation seems to be there
* Testing has been added for relevant portions of code. 

I did leave a few comments on wording in documentation; and a question regarding placement of ancillary code.

File: framework/TSA/ARMA.py (hunk 1 above)
Comment (PaulTalbot-INL) [reply framework/TSA/ARMA.py:65]: I like it. I'll change it.

File: tests/framework/ROM/TimeSeries/SyntheticHistory/TrainingData/generateARMA.py (hunk 2 above)
Comment (PaulTalbot-INL) [reply tests/framework/ROM/TimeSeries/SyntheticHistory/TrainingData/generateARMA.py:1]: The RAVEN run uses this training data, so I think it makes sense to have it here (along with the generated data).

The other option would be to put each generation script and the CSVs in the RAVEN working dirs for each integration test. I don't know which I prefer; I like the grouped similarity of the analytic training data being in a central place.

File: framework/TSA/ARMA.py (hunk 1 above)
Comment (dylanjm) [reply framework/TSA/ARMA.py:65]: Great, just a reminder for the same addition to the "P" parameter above it too.

File: tests/framework/ROM/TimeSeries/SyntheticHistory/TrainingData/generateARMA.py (hunk 2 above)
Comment (dylanjm) [reply tests/framework/ROM/TimeSeries/SyntheticHistory/TrainingData/generateARMA.py:1]: Oh I see, I think I like your first suggestion better, but also like having the analytic stuff in one location. I think making those changes would be better served in a different PR so no hold ups here.

**Review by dylanjm — APPROVED** [review]

**Comment by PaulTalbot-INL** [comment]
The ARMAs on all machines are typically within 1e-4, but there's always some random point that's much further off (2e-2 is the largest I found).

I'm not sure what to think of this yet. I want to test the signals themselves, but that's a pretty large margin of error.

**Review by dylanjm — APPROVED** [review]

**Pull request merged**